### PR TITLE
feat: add random markdown support

### LINKED ISSUE #330: Feature Request: Random Markdown
In my own project, I'm working on a feature that parses markdown. This would be a great feature for that.

### PR DESCRIPTION
**Pull request by techgaun**
closes #330 
I've added:

- [x] USAGE.md docs if applicable
- [x] CHANGELOG.md

### CODE REVIEW

File: lib/faker/markdown.ex [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    |> Enum.join(" ")
+  end
+
+  @doc """
+  Returns a random ordered list
+
+  ## Examples
+
+      iex> Faker.Markdown.ordered_list()
+      "1. Deleniti consequatur et qui vitae et."
```
Comment (anthonator) [inline lib/faker/markdown.ex:65]: Please include at least 4 examples for doc tests.

File: lib/faker/markdown.ex [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    list()
+  end
+
+  @doc """
+  Returns a random unordered list
+
+  ## Examples
+
+      iex> Faker.Markdown.unordered_list()
+      "* Deleniti consequatur et qui vitae et."
```
Comment (anthonator) [inline lib/faker/markdown.ex:78]: Please include at least 4 examples for doc tests.

File: lib/faker/markdown.ex [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    |> Enum.join("\n")
+  end
+
+  @doc """
+  Returns random inline code
+
+  ## Examples
+
+      iex> Faker.Markdown.inline_code()
+      "`Sint deleniti consequatur et qui vitae et quibusdam et sit.`"
```
Comment (anthonator) [inline lib/faker/markdown.ex:103]: Please include at least 4 examples for doc tests.

File: lib/faker/markdown.ex [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    "`#{Lorem.sentence()}`"
+  end
+
+  @doc """
+  Returns random inline code
+
+  ## Examples
+
+      iex> Faker.Markdown.block_code()
+      "```elixir\\nDeleniti consequatur et qui vitae et.\\n```"
```
Comment (anthonator) [inline lib/faker/markdown.ex:116]: Please include at least 4 examples for doc tests.

File: lib/faker/markdown.ex [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    """
+  end
+
+  @doc """
+  Returns random markdown table
+
+  ## Examples
+
+      iex> Faker.Markdown.table()
+      "sint | deleniti | consequatur\\n---- | ---- | ----\\net | qui | vitae\\net | quibusdam | et\\nsit | aut | expedita\\ncumque | est | necessitatibus\\nbeatae | ex | sunt"
```
Comment (anthonator) [inline lib/faker/markdown.ex:133]: Please include at least 4 examples for doc tests.

File: lib/faker/markdown.ex [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    |> Enum.join("\n")
+  end
+
+  @doc """
+  Returns random markdown
+
+  ## Examples
+
+      iex> Faker.Markdown.markdown()
+      "cumque | est | necessitatibus\\n---- | ---- | ----\\nbeatae | ex | sunt\\nsoluta | possimus | soluta\\nasperiores | qui | vitae\\n\\n* Et vitae vitae ut quisquam corporis quisquam ab harum ipsa.\\n* Numquam maxime ut aut inventore eius rerum beatae.\\n* Qui officia vel quaerat expedita.\\n* Perspiciatis rerum nam repellendus inventore nihil.\\n\\n`Sequi ducimus qui voluptates magni quisquam sed odio.`\\n\\n```elixir\\nError non impedit tempora minus voluptatem qui fugit.\\n```\\n\\n### Cupiditate"
```
Comment (anthonator) [inline lib/faker/markdown.ex:162]: Please include at least 4 examples for doc tests.

**Comment by techgaun** [comment]
@anthonator I've added more doc tests now to be 4 on each of them

**Pull request merged**